graph/denota/src/nota-registrar.ts

```diff
@@ -0,0 +1,355 @@
+        : fromAccount;
+    const amount = event.params.amount;
+    const transactionHexHash = event.transaction.hash.toHex();
+    const notaId = event.params.notaId.toString();
+    const transaction = saveTransaction(
+      transactionHexHash,
+      event.block.timestamp,
+      event.block.number
+    );
+  
+    // Load nota
+    let nota = Nota.load(notaId);
+    if (nota == null) {
+      // SHOULDN NEVER BE THE CASE
+      nota = new Nota(notaId);
+      nota.save();
+    }
+    
+
+    // Attaching hook specific parameters
+    // TODO
+
+    const escrow = new Escrow(transactionHexHash + "/" + notaId);
+    escrow.nota = notaId;
+    escrow.caller = fromAccount.id;
+    escrow.to = event.address.toHexString();
+    escrow.amount = amount;
+    escrow.instant = event.params.instant;
+    escrow.moduleFee = event.params.moduleFee
+    escrow.transaction = transaction.id;
+    escrow.save();
+
+    // nota.escrowed += amount;  // TODO does this need to be done or front end handles?
+    // nota.save()
+}
+
+export function handleCashed(event: CashedEvent): void {
+    let toAccount = Account.load(event.params.to.toHexString());
+    toAccount =
+      toAccount == null
+        ? saveNewAccount(event.params.to.toHexString())
+        : toAccount;
+    const amount = event.params.amount;
+    const transactionHexHash = event.transaction.hash.toHex();
+    const transaction = saveTransaction(
+      transactionHexHash,
+      event.block.timestamp,
+      event.block.number
+    );
+    const notaId = event.params.notaId.toString();
+  
+    // Load nota
+    let nota = Nota.load(notaId);
+    if (nota == null) {
+      // SHOULDN'T BE THE CASE
+      nota = new Nota(notaId);
+      nota.save();
+    }
+  
+    // if (nota.moduleData.endsWith("/escrow")) {
+    //   const reversiblePayData = ReversiblePaymentData.load(nota.moduleData);
+    //   if (reversiblePayData) {
+    //     if (nota.owner == toAccount.id) {
+    //       reversiblePayData.status = "RELEASED";
+    //     } else {
+    //       reversiblePayData.status = "VOIDED";
+    //     }
+    //     reversiblePayData.save();
+    //   }
+    // }
+  
+    const escrow = new Escrow(transactionHexHash + "/" + notaId);
+    escrow.caller = event.params.casher.toHexString();
+    escrow.to = toAccount.id;
+    escrow.amount = amount.neg();
+    // escrow.instant = instant;
+    escrow.nota = notaId;
+    // escrow.cashData = event.params.cashData
+    escrow.moduleFee = event.params.moduleFee
+    escrow.transaction = transaction.id;
+    escrow.save();  
+    // nota.escrowed -= amount;  // TODO does this need to be done or front end handles?
+}
+
+// TODO: Transfer event being fired before write event is causing problems
+export function handleTransfer(event: TransferEvent): void {
+  // Load event params
+  const from = event.params.from.toHexString();
+  const to = event.params.to.toHexString();
+  const notaId = event.params.tokenId.toHexString();
```

Comment:
This should be 
`const notaId = event.params.notaId.toString();`

<img width="558" alt="Screenshot 2024-01-31 at 8 13 59 AM" src="https://github.com/denotalabs/denota-protocol/assets/10327933/8a48581c-94f1-4fd2-a498-7e6d7a0915cf">